### ISSUE TITLE
Cursor.columns method fails silently when table name contains high-Unicode character

### ISSUE THREAD

**Issue by gordthompson** [mkleehammer/pyodbc#308]
In my SQL Server 2014 instance I created a table

```sql
CREATE TABLE [dbo].[Team_🍁](
    [id] [int] NOT NULL,
    [txtcol] [nvarchar](50) NULL,
 CONSTRAINT [PK_Team_🍁] PRIMARY KEY CLUSTERED 
(
    [id] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
) ON [PRIMARY]
```

and populated it with a single row

```
id  txtcol
--  ------
 1  foo
```

Using pyodbc 4.0.21 and "ODBC Driver 11 for SQL Server" under Python 3.6 I can successfully query the table

```python
print(crsr.execute("SELECT txtcol FROM [Team_🍁] WHERE id=1").fetchval())
```

and SQL Profiler shows me that the server received

```sql
SELECT txtcol FROM [Team_🍁] WHERE id=1
```

However, when I run the code

```python
num_columns = 0
for fld in crsr.columns('Team_🍁'):
    num_columns += 1
    print(f'  [{fld.column_name}]')
print(f'({num_columns} column(s) found)')
```

I get

```
(0 column(s) found)
```

and SQL Profiler says that the server received

```sql
exec [sys].sp_columns_100 N'Team_ðŸ  ',N'%',NULL,NULL,@ODBCVer=3,@fUsePattern=1
```

where the four (4) characters following `Team_` are `U+F0`, `U+178`, `U+8D`, and `U+81`. For reference the 🍁 character, `U+1F341` is represented in UTF-8 by `F0 9F 8D 81`, which is very close to the code points of the errant four characters above (only the second one is different).

**Comment by gordthompson** [comment]
It also happens with a table named [Team_Ώπα], in which case SQL Profiler shows

```sql
exec [sys].sp_columns_100 N'Team_Î Ï€Î±',N'%',NULL,NULL,@ODBCVer=3,@fUsePattern=1
```

where the six (6) characters following `Team_` are `U+CE U+8F U+CF U+AC U+CE U+B1` and the UTF-8 encoding of Ώπα is `CE 8F CF 80 CE B1`.

**Comment by v-chojas** [comment]
What is the client OS? If it is Linux, what is the unixODBC version?

**Comment by gordthompson** [comment]
@v-chojas 

> What is the client OS?

Windows 7

**Comment by gordthompson** [comment]
Curious to see if there was a reasonably straightforward explanation for the discrepancies between the UTF-8 bytes and the resulting Unicode characters in the above examples, I did a bit of digging. In the following descriptions,  `W+` refers to Windows-1252 characters and `U+` refers to Unicode characters.

#### First example

The Maple Leaf character `U+1F341` is UTF-8 encoded as `F0 9F 8D 81`. Characters in query: `U+F0, U+178, U+8D, and U+81`.

```
W+F0 -> Latin Small Letter Eth -> U+F0
W+9F -> Latin Capital Letter Y With Diaeresis -> U+178
W+8D -> (unused code point) -> U+8D
W+81 -> (unused code point) -> U+81
```

#### Second example

Ώπα is UTF-8 encoded as `CE 8F CF 80 CE B1`. Characters in query: `U+CE U+8F U+CF U+AC U+CE U+B1`.

```
W+CE -> Latin Capital Letter I With Circumflex -> U+CE
W+8F -> (unused code point) -> U+8F
W+CF -> Latin Capital Letter I With Diaeresis -> U+CF
W+80 -> Euro Sign -> U+20AC -> (apparently truncated to) U+AC
W+CE -> Latin Capital Letter I With Circumflex -> U+CE
W+B1 -> Plus-Minus Sign -> U+B1
```

**Comment by v-chojas** [comment]
pyODBC currently uses only narrow API [SQLColumns](https://github.com/mkleehammer/pyodbc/blob/master/src/cursor.cpp#L1298) for cursor.columns(), and passes it a UTF-8 encoded string, which explains what you're seeing:

- On Windows, the narrow APIs interpret strings in the system's codepage, which is likely CP1252 for you. The SQL Server driver internally uses Unicode so the driver manager converts, but from CP1252 to UTF-16, resulting in the misinterpretation you see.
- On Linux, unixODBC 2.3.1 by default assumes the narrow APIs are ASCII-only. You may try [the newer 2.3.4/2.3.5 version](https://github.com/lurcher/unixODBC) of unixODBC which will detect and use the environment's encoding, by default UTF-8, so that may allow it to work.

In summary: a newer unixODBC may let this work on Linux, but the ultimate fix and for Windows is to use SQLColumnsW() in pyODBC.
